src/v3/src/util/idxUtils.ts

```diff
@@ -229,3 +238,90 @@ export const convertIdxMessageToWidgetMessage = (
     message: getMessage(message),
   };
 });
+
+export const convertIdxInputsToRegistrationSchema = (
+  inputs: Input[],
+): RegistrationElementSchema[] => (
+  inputs.reduce((acc: Input[], input: Input) => {
+    const flattenedInputs = flattenInputs(input);
+    return [...acc, ...flattenedInputs];
+  }, []).map((flattendInput: Input) => ({
+    ...flattendInput,
+    'label-top': true,
+    'data-se': flattendInput.name,
+    wide: true,
+  } as RegistrationElementSchema))
+);
+
+export const convertRegistrationSchemaToIdxInputs = (
+  schema: RegistrationElementSchema[],
+  idxInputs: Input[],
+): void => {
+  schema.forEach((schemaEle: RegistrationElementSchema) => {
+    if (schemaEle.name?.includes('.')) {
+      // nested object (only go one level deep)
+      const [schemaGroupName, schemaGrpFieldName] = schemaEle.name.split('.');
+      const inputGroupIndex = idxInputs.findIndex(({ name }) => name === schemaGroupName);
+      if (inputGroupIndex !== -1 && Array.isArray(idxInputs[inputGroupIndex]?.value)) {
+        // Existing input
+        const inputGroupFieldIndex = (
+          idxInputs[inputGroupIndex]?.value as Input[]
+        )?.findIndex(({ name }) => name === schemaGrpFieldName);
+        if (inputGroupFieldIndex !== -1) {
+          // Update existing field
+          const srcObj = (idxInputs[inputGroupIndex]?.value as Input[])?.[inputGroupFieldIndex];
+          Object.assign(srcObj, { ...schemaEle, name: schemaGrpFieldName });
+        } else {
+          // Add new field
+          const schemaInputField = { ...schemaEle, name: schemaGrpFieldName };
+          (idxInputs[inputGroupIndex]?.value as Record<string, unknown>[]).push(schemaInputField);
+        }
+      } else {
+        // New top level input
+        idxInputs.push({
+          name: schemaGroupName,
+          type: 'object',
+          required: schemaEle.required,
+          // @ts-expect-error Registration Schema Element has extra fields that are not in Input
+          value: { ...schemaEle, name: schemaGrpFieldName },
+        });
+      }
+    } else {
+      // not nested
+      const existFieldIdx = idxInputs.findIndex(({ name }) => name === schemaEle.name);
+      if (existFieldIdx !== -1) {
+        // Existing Field (modify)
+        Object.assign(idxInputs[existFieldIdx], schemaEle);
+      } else {
+        // New Field (create)
+        // @ts-expect-error Registration Schema Element has extra fields that are not in Input
+        idxInputs.push(schemaEle);
+      }
+    }
+  });
+};
```

Comment:
This function converts the flattened Registration schema back into an `Input` object array. There is an assumption made here that we wont nest objects more than 1 level (because testing in v2, those properties are ignore when I tried them).